### PR TITLE
Add workflow for weekly build and publish of snap

### PR DESCRIPTION
**Pull request by jpm-canonical**
## Summary
The github workflow builds, tests and publishes the snap. Build is done in Qemu on amd64, building only for arm64. Test is done on native arm64. Publish is done on amd64. The workflow is triggered weekly to pull in latest matter sdk.

## Testing Steps

Check github action output of this PR, or after it is merged the output of the weekly triggered action.

### CODE REVIEW

**Comment by farshidtz** [comment]
Supersedes #66

**Comment by jpm-canonical** [comment]
Test is failing with this error:
```
[SETUP]
[exec] ./gpio-mock.sh
[exec] rm -rf squashfs-root
[exec] unsquashfs ..//home/runner/work/matter-pi-gpio-commander/matter-pi-gpio-commander/matter-pi-gpio-commander_2.0.0_arm64.snap
stderr: Could not open ..//home/runner/work/matter-pi-gpio-commander/matter-pi-gpio-commander/matter-pi-gpio-commander_2.0.0_arm64.snap, because No such file or directory
err: exit status 1
```

File: .github/workflows/build-and-test-snap.yml [hunk 1]
```diff
@@ -31,8 +64,9 @@ jobs:
         env:
           MOCK_GPIO: true
           SKIP_TEARDOWN_REMOVAL: true
-          LOCAL_SERVICE_SNAP: ../${{ steps.snapcraft.outputs.snap }}
-        run: go test -failfast -p 1 -timeout 20m -v
+          LOCAL_SERVICE_SNAP: ../${{ needs.build-arm64.outputs.snap }}
```
Comment (locnnil) [inline .github/workflows/build-and-test-snap.yml]: You're referring it using the environment variable which gives you an absolute path, removing `../` should fix it: 

```suggestion
          LOCAL_SERVICE_SNAP: ${{ needs.build-arm64.outputs.snap }}
```

**Review by locnnil — CHANGES_REQUESTED** [review]

**Review by locnnil — COMMENTED** [review]
Now [the problem](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/9522778831/job/26254542770#step:5:40) is actually related with the gpio mockup module.

**Pull request closed without merging**